tests/integration/modules/mac_system.py

```diff
@@ -0,0 +1,184 @@
+'''
+
+# Import python libs
+from __future__ import absolute_import
+import random
+import string
+
+# Import Salt Testing libs
+from salttesting.helpers import ensure_in_syspath, destructiveTest
+from salt.ext.six.moves import range
+ensure_in_syspath('../../')
+
+# Import salt libs
+import integration
+import salt.utils
+
+ATRUN_ENABLED = False
+REMOTE_LOGIN_ENABLED = False
+REMOTE_EVENTS_ENABLED = False
+COMPUTER_NAME = ''
+SUBNET_NAME = ''
+KEYBOARD_DISABLED = False
+SET_COMPUTER_NAME = __random_string()
+SET_SUBNET_NAME = __random_string()
+
+
+def __random_string(size=6):
+    '''
+    Generates a random username
+    '''
+    return 'RS-' + ''.join(
+        random.choice(string.ascii_uppercase + string.digits)
+        for x in range(size)
+    )
+
+
+class MacSystemModuleTest(integration.ModuleCase):
+    '''
+    Validate the mac_system module
+    '''
+
+    def setUp(self):
+        '''
+        Get current settings
+        '''
+        if not salt.utils.is_darwin():
+            self.skipTest('Test only available on Mac OS X')
+
+        if not salt.utils.which('systemsetup'):
+            self.skipTest('Test requires systemsetup binary')
+
+        if not salt.utils.which('launchctl'):
+            self.skipTest('Test requires launchctl binary')
+
+        if salt.utils.get_uid(salt.utils.get_user()) != 0:
+            self.skipTest('Test requires root')
+
+        ATRUN_ENABLED = self.run_function('service.enabled',
+                                          ['com.apple.atrun'])
+        REMOTE_LOGIN_ENABLED = self.run_function('system.get_remote_login')
+        REMOTE_EVENTS_ENABLED = self.run_function('system.get_remote_events')
+        COMPUTER_NAME = self.run_function('system.get_computer_name')
+        SUBNET_NAME = self.run_function('system.get_subnet_name')
+        KEYBOARD_DISABLED = self.run_function(
+            'system.get_disable_keyboard_on_lock')
+
+        super(MacSystemModuleTest, self).setUp()
+
+    def tearDown(self):
+        '''
+        Reset to original settings
+        '''
+        if not ATRUN_ENABLED:
+            atrun = '/System/Library/LaunchDaemons/com.apple.atrun.plist'
+            self.run_function('service.stop', [atrun])
+
+        self.run_function('system.set_remote_login', [REMOTE_LOGIN_ENABLED])
+        self.run_function('system.set_remote_events', [REMOTE_EVENTS_ENABLED])
+        self.run_function('system.set_computer_name', [COMPUTER_NAME])
+        self.run_function('system.set_subnet_name', [SUBNET_NAME])
+        self.run_function('system.set_disable_keyboard_on_lock',
+                          [KEYBOARD_DISABLED])
+
+        super(MacSystemModuleTest, self).tearDown()
+
+    @destructiveTest
+    def test_get_set_remote_login(self):
+        '''
+        Test system.get_remote_login
+        Test system.set_remote_login
+        '''
+        self.run_function('system.set_remote_login', [True])
+        self.assertTrue(self.run_function('system.get_remote_login'))
```

Comment:
This function appears to _always_ return True indicating that the command was run, versus actually looking at the actual return from the shell command. This may be a bug.